teachertool/src/transforms/handleProjectOnLoadAsync.ts

```diff
@@ -0,0 +1,32 @@
+import { Ticks } from "../constants";
+import { stateAndDispatch } from "../state";
+import { loadChecklistAsync } from "./loadChecklistAsync";
+import { loadProjectMetadataAsync } from "./loadProjectMetadataAsync";
+import { setActiveTab } from "./setActiveTab";
+
+/**
+ * The user may use optional URL parameters to specify a default project to load when opening the eval tool.
+ * This function handles loading that project and the additional on-load behavior that occurs as a result.
+ */
+export async function handleProjectOnLoadAsync(defaultChecklistUrl?: string | undefined): Promise<void> {
+    const { state } = stateAndDispatch();
+
+    const projectParam = window.location.href.match(/project=([^&]+)/)?.[1];
+    if (!!projectParam) {
+        const decoded = decodeURIComponent(projectParam);
+        const shareId = pxt.Cloud.parseScriptId(decoded);
+        if (!!shareId) {
+            pxt.tickEvent(Ticks.LoadProjectFromUrl);
```

Comment:
If would you prefer to remove the `project` param from the URL after processing it, this can be accomplished without a page reload using `window.history.replaceState`.